src/dumch/dartclojure.cljc

```diff
@@ -0,0 +1,114 @@
+    :default false
+    :parse-fn #?(:clj #(Boolean/parseBoolean %)
+                 :cljs read-string)
+    :validate [boolean? "Must be either true or false"]]
+   ["-e" "--end TEXT" "Mode: text or symbol to show the end of code"
+    :default ""
+    :parse-fn str]
+   ["-m" "--material MATERIAL" "material require alias, any string"
+    :default "m"
+    :parse-fn str
+    :validate [#(re-matches #"^[A-Za-z][A-Za-z0-9]*$" %)
+               "invalid require alias for material"]]
+   ["-f" "--flutter FLUTTER_MACRO_PACKAGE" "flutter-macro require alias, any string"
+    :default "f"
+    :parse-fn str
+    :validate [#(re-matches #"^[A-Za-z][A-Za-z0-9]*$" %)
+               "invalid require alias for flutter-materail"]]
+   ["-c" "--colors BOOLEAN" "colorize output"
+    :default false
+    :parse-fn #?(:clj #(Boolean/parseBoolean %)
+                 :cljs read-string)
+    :validate [boolean? "Must be either true or false"]]
+   ["-h" "--help"]])
+
+(defn show! [data & {:keys [colors] :or {colors false}}]
+  (if colors
+    (zpr/czprint data {:parse-string? true})
+    (println data)))
+
+(defn- print-repl-instructions [end]
+  (println (str "Paste dart code below, press enter"
+                (when (seq end) (str ", write '" end "',"))
+                " and see the result:\n")))
+
+#?(:clj
+   (defn- stdin-loop! [& {:keys [material flutter end colors]}]
+     (print-repl-instructions end)
+     (loop [input (read-line)
+            acc []]
+       (if (= input end)
+         (do
+           (try
+             (-> (convert (str/join "\n" acc)
+                          :material material
+                          :flutter flutter)
+                 (show! :colors colors))
+             (catch Exception e (println "Can't convert the code; e " e)))
+           (println)
+           (recur (read-line) []))
+         (recur (read-line) (conj acc input))))))
+
+#?(:cljs
+   ;; This could have been a lambda, but then we can't update it via the REPL
+   (defn- process-node-repl-input! [input !code {:keys [material flutter end colors]}]
+     (if (= input end)
+       (do
+         (try
+           (-> (convert (str/join "\n" @!code)
+                        :material material
+                        :flutter flutter)
+               (show! :colors colors))
+           (catch js/Error e (println "Can't convert the code; e " e)))
+         (println)
+         (reset! !code []))
+       (swap! !code conj input))))
+
+#?(:cljs
+   (defn- node-repl! [& {:keys [end] :as params}]
+     (print-repl-instructions end)
+     (let [!code (atom [])]
+       (-> (.createInterface node-rl #js {:input js/process.stdin
+                                          :output js/process.stdout
+                                          :terminal false})
+           (.on "line" (fn [input]
+                         (process-node-repl-input! input !code params)))))))
+
+(defn -main [& args]
+  (let [{{:keys [repl help] :as params} :options,
```

Comment:
in the main: `:keys [repl help path]`

and cond branch:
```clojure
      path (if-let [code (slurp path)]
             (apply show! 
                    (apply convert code params)
                    params))
```

Reply:
✅